cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -661,13 +687,35 @@ func main() {
 			}
 		}
 
-		if capacityController != nil {
-			go capacityController.Run(ctx, int(*capacityThreads))
-		}
-		if csiClaimController != nil {
-			go csiClaimController.Run(ctx, int(*finalizerThreads))
+		if utilfeature.DefaultFeatureGate.Enabled(features.ReleaseLeaderElectionOnExit) {
+			var wg sync.WaitGroup
+			if capacityController != nil {
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					capacityController.Run(controllerCtx, int(*capacityThreads), &wg)
+				}()
+			}
+			if csiClaimController != nil {
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					csiClaimController.Run(controllerCtx, int(*finalizerThreads), &wg)
+				}()
+			}
+			provisionController.ControllerWaitGroup(&wg)
```

Comment:
Ouch, this should have been an `option` in `NewProvisionController()`. Too late to change this, I already tagged the lib-external-provisioner with this function.
Note to self: review external-provisioner PRs before merging things to the provisioner lib.